test/unit/request.test.js

```diff
@@ -581,6 +617,100 @@ describe('Request', function () {
+                            value: 'otherheader',
+                            disabled: true
+                        }
+                    ],
+                    auth: {
+                        type: 'basic',
+                        username: 'foo',
+                        password: 'bar'
+                    }
+                },
+                request = new Request(rawRequest);
+
+            expect(request).to.have.property('auth');
+            expect(request.toJSON()).to.have.property('auth');
+            request.authorizeUsing(null);
+
+            expect(request).to.not.have.property('auth');
+            expect(request.toJSON()).to.not.have.property('auth');
+            expect(request.auth).be.undefined;
+        });
+
+        it('should not update auth if given type is invalid', function () {
+            var rawRequest = {
+                    url: 'postman-echo.com',
+                    method: 'GET',
+                    header: [
+                        {
+                            key: 'some',
+                            value: 'header'
+                        },
+                        {
+                            key: 'other',
+                            value: 'otherheader',
+                            disabled: true
+                        }
+                    ],
+                    auth: {
+                        type: 'basic',
+                        basic: [{
+                            key: 'username',
+                            type: 'string',
+                            value: 'postman'
+                        },
+                        {
+                            key: 'password',
+                            type: 'string',
+                            value: 'password'
+                        }]
+                    }
+                },
+                request = new Request(rawRequest);
+
+            expect(request).to.have.property('auth');
+            expect(request.toJSON()).to.have.property('auth');
+            expect(request.auth.toJSON()).to.be.eql({
+                'basic': [{
+                    'key': 'username',
+                    'type': 'string',
+                    'value': 'postman'
+                }, {
+                    'key': 'password',
+                    'type': 'string',
+                    'value': 'password'
+                }],
+                'type': 'basic'
+            });
+            request.authorizeUsing('type');
+
+            // after setting to invalid type, the method should return with no change
+            expect(request.auth.toJSON()).to.be.eql({
+                'basic': [{
+                    'key': 'username',
+                    'type': 'string',
+                    'value': 'postman'
+                }, {
+                    'key': 'password',
+                    'type': 'string',
+                    'value': 'password'
+                }],
+                'type': 'basic'
+            });
+        });
```

Comment:
Then why's this test here and not in #1101?

Reply:
This check is for `requests.js` method `authorizeUsing` method, https://github.com/postmanlabs/postman-collection/pull/1101 is for `request-auth.js` method `update`.